cou/cli.py

```diff
@@ -91,11 +91,12 @@ async def analyze_and_plan(model_name: Optional[str] = None) -> UpgradeStep:
     :return: Generated upgrade plan.
     :rtype: UpgradeStep
     """
-    model_name = model_name or await juju_utils.get_current_model_name()
+    model = COUModel(model_name)
+    await model.check_model_name()  # check model name and obtain if it was None
```

Comment:
I'm not a fan of the pattern here. Why do we want to call `check_model_name` outside the class? Wouldn't it be better to let COUModel automatically handle the situation where model_name input is `None`?

Also, maybe we should think about whether to auto-connect to a model when instantiating a COUModel.

Reply:
We cannot run async functions in `__init__`.

Reply:
Alternative would be something like this:
```
    @classmethod
    async def create(cls, name: str):
        model = COUModel(name)
        await model.check_model_name()
        return await model._connect()
```

Reply:
We can get even loop and run it on that, but I'm not sure how it will behave or we can go with Erhan suggestion or simply define `create_model` function outside.

Reply:
As discussed elsewhere, we agreed on Erhan's suggestion with one modificaton: remove `check_model_name` method.

Reply:
I added the staticmethod create  with simple `_connect` call and remove `check_model_name`.

Reply:
Looks good, thanks!